### PR TITLE
AMP hosted article page

### PR DESCRIPTION
**Pull request by Calanthe**
This PR introduces AMP version of the hosted article page.
Other pages (video & gallery) are going to be done later.

## Screenshots
screen shot 2016-11-17 at 17 06 24

screen shot 2016-11-17 at 17 06 30

screen shot 2016-11-17 at 17 06 40

## Request for comment
@NataliaLKB @regiskuckaertz

### CODE REVIEW

**Comment by regiskuckaertz** [comment]
woohoo 😮 

would be super interested to learn how you made the popup, because I'd need it too for the glabs traffic driver 😄 

will review tomorrow

File: commercial/app/views/hosted/guardianHostedHeader.scala.html [hunk 1]
```diff
@@ -1,17 +1,48 @@
 @import common.commercial.hosted.HostedPage
-@(pageCssClass: String, page: HostedPage)
+@(pageCssClass: String, page: HostedPage, isAMP: Boolean = false)
```
Comment (kelvin-chappell) [inline commercial/app/views/hosted/guardianHostedHeader.scala.html]: Could get rid of default value.

File: article/test/ArticleAmpValidityTest.scala [hunk 2]
```diff
@@ -17,6 +17,7 @@ import org.scalatest.DoNotDiscover
     "/football/2016/jul/10/france-portugal-euro-2016-match-report", // Match summary
     "/us-news/live/2016/aug/12/donald-trump-republicans-hillary-clinton-us-election-live", // Live blog
     "/sport/live/2016/aug/20/rio-2016-olympics-day-15-mo-farah-relays-tom-daley-nicola-adams-football-live", // Sport live blog
-    "/somerset-county-council-partner-zone/2016/sep/21/somerset-offers-opportunities-develop-social-work-career" // Paid for by hosted page
+    "/somerset-county-council-partner-zone/2016/sep/21/somerset-offers-opportunities-develop-social-work-career", // Paid for by hosted page
+    "/advertiser-content/audi-history-of-audi/audi-and-innovation" // Hosted article page
```
Comment (kelvin-chappell) [inline article/test/ArticleAmpValidityTest.scala]: Probably best not to include this test until we've got it all valid.

File: tools/amp-validation/endpoints/fixed.js [hunk 3]
```diff
@@ -61,6 +61,11 @@ module.exports = (() => {
         /*
             Paid for by hosted pages
          */
-        '/somerset-county-council-partner-zone/2016/sep/21/somerset-offers-opportunities-develop-social-work-career'
+        '/somerset-county-council-partner-zone/2016/sep/21/somerset-offers-opportunities-develop-social-work-career',
+
+        /*
+            Hosted article page
+         */
+        '/advertiser-content/audi-history-of-audi/audi-and-innovation'
```
Comment (kelvin-chappell) [inline tools/amp-validation/endpoints/fixed.js]: Probably best not to include this test until page is completely valid.

File: tools/amp-validation/endpoints/fixed.js [hunk 4]
```diff
@@ -61,6 +61,11 @@ module.exports = (() => {
         /*
             Paid for by hosted pages
```
Comment (kelvin-chappell) [inline tools/amp-validation/endpoints/fixed.js]: These aren't hosted pages.

File: article/test/ArticleAmpValidityTest.scala [hunk 5]
```diff
@@ -17,6 +17,7 @@ import org.scalatest.DoNotDiscover
     "/football/2016/jul/10/france-portugal-euro-2016-match-report", // Match summary
     "/us-news/live/2016/aug/12/donald-trump-republicans-hillary-clinton-us-election-live", // Live blog
     "/sport/live/2016/aug/20/rio-2016-olympics-day-15-mo-farah-relays-tom-daley-nicola-adams-football-live", // Sport live blog
-    "/somerset-county-council-partner-zone/2016/sep/21/somerset-offers-opportunities-develop-social-work-career" // Paid for by hosted page
+    "/somerset-county-council-partner-zone/2016/sep/21/somerset-offers-opportunities-develop-social-work-career", // Paid for by hosted page
```
Comment (kelvin-chappell) [inline article/test/ArticleAmpValidityTest.scala]: This is just a paid-for page; it isn't hosted.

**Review by kelvin-chappell — APPROVED** [review]
With some minor issues, looks good! 👍 
Once this is merged I'll add my image fix.

File: common/app/views/fragments/amp/stylesheets/hosted.scala.html [hunk 6]
```diff
@@ -0,0 +1,734 @@
...
+}
+
+.hosted-tone-btn {
+    color: @brandColour;
+    border: 1px solid @brandColour;
+}
+
+.hosted__banner .hosted-tone-btn,
+.hosted-tone-btn:focus,
+.hosted-tone-btn:hover {
```
Comment (SiAdcock) [inline common/app/views/fragments/amp/stylesheets/hosted.scala.html]: Is `:hover` useful on mobile devices?

**Comment by Calanthe** [comment]
@regiskuckaertz The popup is really easy. You just need a `checkbox input` (which is hidden), a `label` (which is styled as a button/link) and a popup `div` which ideally is located as a sibling of the aforementioned `input`:

```
<input type="checkbox" value="" id="hostedAboutTop" class="paidfor-input">
<label for="hostedAboutTop" class="tooltip-label">About</label>
<div class="survey-container">Popup info here</div>
```

Then in CSS, after clicking on the `label` (and checking the `checkbox input`), all you need to do is show/hide the popup:

```
.paidfor-input:checked ~ .survey-container {
    display: block;
}

.paidfor-input:not(:checked) ~ .survey-container {
    display: none;
} 
```

I even made a game based on this mechanism (http://zofiakorcz.pl/case-study-duck-hunt) ;)

File: commercial/app/views/hosted/guardianHostedHeader.scala.html (hunk 1 above)
Comment (Calanthe) [inline commercial/app/views/hosted/guardianHostedHeader.scala.html]: But then I will have to pass this value everywhere where the header is called.

**Comment by regiskuckaertz** [comment]
@Calanthe total mic drop there Zofia! Thanks a lot, this is exactly what I needed 😃 

Great work!

**Comment by regiskuckaertz** [comment]
Will read that post this weekend :-)

File: commercial/app/views/hosted/guardianHostedHeader.scala.html (hunk 1 above)
Comment (kelvin-chappell) [inline commercial/app/views/hosted/guardianHostedHeader.scala.html]: Yes.  It looks like a safe default, so ignore comment.

**Comment by kelvin-chappell** [comment]
Image fix is in PR #15042

**Pull request merged**